streams/src/test/java/org/apache/kafka/streams/integration/AbstractJoinIntegrationTest.java

```diff
@@ -88,31 +88,35 @@ public static Collection<Object[]> data() {
     StreamsBuilder builder;
 
     private final List<Input<String>> input = Arrays.asList(
-        new Input<>(INPUT_TOPIC_LEFT, null),
-        new Input<>(INPUT_TOPIC_RIGHT, null),
-        new Input<>(INPUT_TOPIC_LEFT, "A"),
-        new Input<>(INPUT_TOPIC_RIGHT, "a"),
-        new Input<>(INPUT_TOPIC_LEFT, "B"),
-        new Input<>(INPUT_TOPIC_RIGHT, "b"),
-        new Input<>(INPUT_TOPIC_LEFT, null),
-        new Input<>(INPUT_TOPIC_RIGHT, null),
-        new Input<>(INPUT_TOPIC_LEFT, "C"),
-        new Input<>(INPUT_TOPIC_RIGHT, "c"),
-        new Input<>(INPUT_TOPIC_RIGHT, null),
-        new Input<>(INPUT_TOPIC_LEFT, null),
-        new Input<>(INPUT_TOPIC_RIGHT, null),
-        new Input<>(INPUT_TOPIC_RIGHT, "d"),
-        new Input<>(INPUT_TOPIC_LEFT, "D")
+        new Input<>(INPUT_TOPIC_LEFT, null, 1),
+        new Input<>(INPUT_TOPIC_RIGHT, null, 2),
+        new Input<>(INPUT_TOPIC_LEFT, "A", 3),
+        new Input<>(INPUT_TOPIC_RIGHT, "a", 4),
+        new Input<>(INPUT_TOPIC_LEFT, "B", 5),
+        new Input<>(INPUT_TOPIC_RIGHT, "b", 6),
+        new Input<>(INPUT_TOPIC_LEFT, null, 7),
+        new Input<>(INPUT_TOPIC_RIGHT, null, 8),
+        new Input<>(INPUT_TOPIC_LEFT, "C", 9),
+        new Input<>(INPUT_TOPIC_RIGHT, "c", 10),
+        new Input<>(INPUT_TOPIC_RIGHT, null, 11),
+        new Input<>(INPUT_TOPIC_LEFT, null, 12),
+        new Input<>(INPUT_TOPIC_RIGHT, null, 13),
+        new Input<>(INPUT_TOPIC_RIGHT, "d", 14),
+        new Input<>(INPUT_TOPIC_LEFT, "D", 15),
+        new Input<>(INPUT_TOPIC_LEFT, "E", 4), // out-of-order data
```

Comment:
Seems we only need this for table-table join tests, but it only blows up stream-stream join results? I am ok to merge as is, but seems it's adding "some noise" the stream-stream-join case (not too bad for binary joins, but for 3-way join it's a lot).

Not sure if we could have different inputs for stream vs table case?

Reply:
Yeah, I noticed this as well. If you don't think having the out-of-order data in stream-stream join tests is valuable, I can open a follow-up PR to remove this (to reduce noise). Just LMK.

Reply:
Fixed in https://github.com/apache/kafka/pull/13609.